Mathlib/Order/Interval/Finset/Nat.lean

```diff
@@ -120,84 +120,127 @@ theorem card_fintypeIic : Fintype.card (Set.Iic b) = b + 1 := by
 -- @[simp]
 theorem card_fintypeIio : Fintype.card (Set.Iio b) = b := by rw [Fintype.card_ofFinset, card_Iio]
 
--- TODO@Yaël: Generalize all the following lemmas to `SuccOrder`
```

Comment:
All of these lemmas already have `SuccOrder` versions, right? I think it'd make sense to have a [`SuccAddOrder`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Order/SuccPred.html#SuccAddOrder) version of these lemmas and state them in that generality, instead of only having them for `Nat`.

Reply:
I agree we shouldn't have them only for Nat (although we should have them for Nat for rewriting!). I'll let Yaël decide on whether they should have SuccAddOrder versions, but if so then that can be a different PR

Reply:
cc @YaelDillies

Reply:
Yes, this API is precisely my original motivation for wanting `SuccAddOrder`

Reply:
This doesn't need to be in this PR - but the TODO shouldn't be removed, perhaps instead adjusted to `SuccAddOrder`.

Reply:
`SuccAddOrder` now exists. Can you update this PR to use it?